### DISCUSSION TITLE
Why is `admin:org` scope required to read organization public key?

### DISCUSSION THREAD

**Discussion by dnrce** [community/community#41919] [Actions]
### Select Topic Area

Product Feedback

### Body

I'd like a workflow that allows non-admins to encrypt secrets that they can provide, encrypted, to admins who can then add those secrets to a repo. This would allow for a more self-service model without extending admin access too broadly.

There is a straightforward answer to how this could be done: use the organization's public key to encrypt the secret (as documented [here](https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#create-or-update-an-organization-secret)) before providing it to the admin.

However, the endpoint to `GET` an organization's public key requires `admin:org` permissions ([ref](https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28#get-an-organization-public-key)). This effectively makes it as hard to encrypt a secret as it is to upload the secret.

Why is this? This is a public key. Is there a reason it cannot be made available to non-admins?

**Comment by DSDmark** [comment]
Hi @dnrce,

the public key is an important part of the organization's security infrastructure. It is used to encrypt secrets, which are then stored in the organization's repository. It would increase the risk of unauthorized access to the organization's secrets If the public key were made available to non-admins. You can use a different workflow to allow non-admins to provide encrypted secrets to admins.

**Reply by dnrce** [reply]
This is asymmetric cryptography, and it's right there in the name... it's a _public_ key. If there is something about the implementation that makes the public key unsafe to share, that would be worrisome.

**Reply by DSDmark** [reply]
It's worth noting that the public key itself is not a secret, and it does not need to be protected in the same way as the private key. However, the security of the organization's secrets depends on the proper use of the public key, so it is important to manage access to it carefully.

**Reply by DSDmark** [reply]
The reason that access to the public key might be restricted in some cases

**Reply by dnrce** [reply]
> However, the security of the organization's secrets depends on the proper use of the public key, so it is important to manage access to it carefully.

Can you explain this further, or explain how allowing other credentialed non-admins to access the public key would _not_ be careful management? Without specifics this seems like FUD.
